specification/cognitiveservices/data-plane/Face/v1.0/Face.json

```diff
@@ -0,0 +1,2655 @@
+     "application/json"
+    ]
+   }
+  },
+  "/persongroups/{personGroupId}/training": {
+   "get": {
+    "description": "Retrieve the training status of a person group (completed or ongoing).",
+    "operationId": "PersonGroups_GetTrainingStatus",
+    "parameters": [
+     {
+      "name": "personGroupId",
+      "in": "path",
+      "description": "personGroupId of target person group.",
+      "required": true,
+      "type": "string"
+     },
+     {
+      "$ref": "#/parameters/SubscriptionKey"
+     }
+    ],
+    "responses": {
+     "200": {
+      "description": "A successful call returns the person group's training status.",
+      "schema": {
+       "$ref": "#/definitions/TrainingStatus"
+      },
+      "examples": {
+       "application/json": {
+        "status": "succeeded",
+        "createdDateTime": "1/3/2017 4:11:35 AM",
+        "lastActionDateTime": null,
+        "message": null
+       }
+      }
+     },
+     "default": {
+      "description": "Error response.",
+      "schema": {
+       "$ref": "#/definitions/APIError"
+      }
+     }
+    },
+    "produces": [
+     "application/json"
+    ]
+   }
+  },
+  "/persongroups": {
+   "get": {
+    "description": "List person groups and their information.",
+    "operationId": "PersonGroups_List",
+    "parameters": [
+     {
+      "name": "start",
+      "in": "query",
+      "description": "List person groups from the least personGroupId greater than the \"start\". It contains no more than 64 characters. Default is empty. ",
+      "type": "string",
+      "enum": [
+       ""
+      ]
+     },
+     {
+      "name": "top",
+      "in": "query",
+      "description": "The number of person groups to list, ranging in [1, 1000]. Default is 1000. ",
+      "type": "integer",
+      "default": "1000",
+      "enum": [
+       "1000"
+      ]
+     },
+     {
+      "$ref": "#/parameters/SubscriptionKey"
+     }
+    ],
+    "responses": {
+     "200": {
+      "description": "A successful call returns an array of person groups and their information (personGroupId, name and userData).",
+      "schema": {
+       "$ref": "#/definitions/ListPersonGroupResponse"
+      },
+      "examples": {
+       "application/json": [
+        {
+         "personGroupId": "sample_group",
+         "name": "group1",
+         "userData": "User-provideddataattachedtothepersongroup"
+        },
+        {
+         "personGroupId": "sample_group2",
+         "name": "group2",
+         "userData": "User-provideddataattachedtothepersongroup"
+        }
+       ]
+      }
+     },
+     "default": {
+      "description": "Error response.",
+      "schema": {
+       "$ref": "#/definitions/APIError"
+      }
+     }
+    },
+    "produces": [
+     "application/json"
+    ]
+   }
+  },
+  "/persongroups/{personGroupId}/train": {
+   "post": {
+    "description": "Queue a person group training task, the training task may not be started immediately.",
+    "operationId": "PersonGroups_Train",
```

Comment:
I'm curious about your `operationId`s.

The generator will group the methods into classes based on what's in front of the underscore, and method names from what is after.

The classes I see getting generated are 

``` csharp

Faces
PersonGroups
Persons
FaceLists

```

Are these the names of the classes that you'd like to see? 

Some code would look like:

``` csharp

var api = new FaceAPI ();

IList<DetectResponseElement> results = api.Faces.DetectAsync( ... ) 

```

Is `Faces` the  right word there? Maybe `Face`? 
And the result type is `DetectResponseElement` .. and you get an `IList<>` of those.  -- could we find a better word like `Detection` (or something that describes a purpose rather than some `Element`

(If I'm way off base here, we can discuss it more ...)